packages/desktop-client/src/components/schedules/ScheduleLink.tsx

```diff
@@ -4,24 +4,35 @@ import React, { useCallback, useRef, useState } from 'react';
 import { useSchedules } from 'loot-core/src/client/data-hooks/schedules';
 import { send } from 'loot-core/src/platform/client/fetch';
 import { type Query } from 'loot-core/src/shared/query';
+import { type TransactionEntity } from 'loot-core/src/types/models';
 
 import { type BoundActions } from '../../hooks/useActions';
+import { SvgAdd } from '../../icons/v0';
 import { type CommonModalProps } from '../../types/modals';
+import { Button } from '../common/Button';
 import { Modal } from '../common/Modal';
 import { Search } from '../common/Search';
 import { Text } from '../common/Text';
 import { View } from '../common/View';
 
 import { ROW_HEIGHT, SchedulesTable } from './SchedulesTable';
 
+type ModalParams = {
+  id: string;
+  transaction: TransactionEntity;
+};
 export function ScheduleLink({
   modalProps,
   actions,
   transactionIds: ids,
+  getTransaction,
+  pushModal,
 }: {
   actions: BoundActions;
   modalProps?: CommonModalProps;
   transactionIds: string[];
+  getTransaction: (a: string) => TransactionEntity;
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏what's `a`? We should be more explicit
```suggestion
  getTransaction: (id: string) => TransactionEntity;
```

Reply:
Fair cop, Was being lazy